docs/contributing.md

```diff
@@ -0,0 +1,58 @@
+# Contributing
+## Introduction
+Hubcast welcomes contributions via [Pull Requests](https://github.com/LLNL/hubcast/pulls).
+We've labeled beginner friendly (good first issue) tasks in the issue tracker, feel free
+to reach out and ask for help when getting started.
+
+For small changes (e.g. bug fixes), feel free to submit a PR.
+
+For larger architectual changes and new features, consider opening an
+[issue](https://github.com/LLNL/hubcast/issues/new?template=issue-feature-request.md) outlining your
+proposed contribution.
+
+## Prerequisites
+Hubcast is written in python. You'll need a version of python and pip to
+install the required dependencies and nodejs to install the
+[smee-client](https://www.npmjs.com/package/smee-client) to test the application locally.
+
+You can install the full development environment using [Spack](spack-develop.md).
+
+## Development
+After cloning the repository you'll need to follow the [Getting Started](getting-started.md)
+documentation to setup a testing,
+1. GitHub Repo
+2. GitHub App
+3. GitLab Repo
+4. GitLab Repo Webhook
+5. GitLab Repo Access Token
+
+> [!TIP]
+> If you're developing locally you can use [smee.io](https://smee.io) to relay
```

Comment:
Should we add `smee` as [an optional dependency](https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#dependencies-optional-dependencies)?

Reply:
Haha no worries 😂